lib/spack/spack/directives.py

```diff
@@ -259,6 +262,29 @@ def variant(pkg, name, default=False, description=""):
     pkg.variants[name] = Variant(default, description)
 
 
+@directive('resources')
+def resource(pkg, **kwargs):
+    """
+    Define an external resource to be fetched and staged when building the package. Based on the keywords present in the
+    dictionary the appropriate FetchStrategy will be used for the resource.
+
+    List of recognized keywords:
+
+    * 'when' : represents the condition upon which the resource is needed (optional)
+    * 'destination' : path where to extract / checkout the resource (optional)
+
+    """
+    when = kwargs.get('when', pkg.name)
+    # FIXME : currently I assume destination to be a relative path (rooted at pkg.stage.source_path)
+    destination = kwargs.get('destination', "")
```

Comment:
I think assuming a relative path is reasonable -- should we even allow absolute paths here?  You might want to raise an exception from the directive if the user even provides an absolute path.

The only other case that might be useful would be if the user wanted to fetch something additional and NOT copy it into the source.  For that it might be useful to give package authors a way to access the path to the resource's stage.  Since resources have names, that should be similar to the way homebrew does it, e.g. we could put a `resources` object in module scope and allow something like this in `install()`:

```python
    def install(self, spec, prefix):
        with working_dir(resources.resource_name.path):
            # do stuff in the resource directory...
```

Where `path` is just a property that returns the stage's `source_dir`.